backend/gsr_booking/api_wrapper.py

```diff
@@ -234,117 +132,37 @@ def get_availability(self, lid, start, end, username):
                     valid_slots.append(slot)
                 room["availability"] = valid_slots
         return rooms
-
-    def book_room(self, rid, start, end, user, lid):
-        """Books room if pennkey is valid"""
-
-        start_date = datetime.datetime.strptime(start, "%Y-%m-%dT%H:%M:%S%z")
-        end_date = datetime.datetime.strptime(end, "%Y-%m-%dT%H:%M:%S%z")
-        duration = int((end_date.timestamp() - start_date.timestamp()) / 60)
-
-        if self.check_credits(user)[lid] < duration:
-            raise APIError("Not Enough Credits to Book")
-
-        payload = {
-            "start": start,
-            "end": end,
-            "pennkey": user.username,
-            "room": rid,
-        }
-        url = f"{WHARTON_URL}{user.username}/student_reserve"
-        response = self.request("POST", url, json=payload).json()
-        if "error" in response:
-            raise APIError("Wharton: " + response["error"])
-        return response
-
+    
     def get_reservations(self, user):
-        booking_ids = set()
-
-        reservations = Reservation.objects.filter(
-            creator=user, end__gte=timezone.localtime(), is_cancelled=False
-        )
-        group_gsrs = GSRBooking.objects.filter(
-            reservation__in=reservations, gsr__in=GSR.objects.filter(kind=GSR.KIND_WHARTON)
-        )
-
-        wharton_bookings = GSRBookingSerializer(
-            GSRBooking.objects.filter(
-                user=user,
-                gsr__in=GSR.objects.filter(kind=GSR.KIND_WHARTON),
-                end__gte=timezone.localtime(),
-                is_cancelled=False,
-            ).union(group_gsrs),
-            many=True,
-        ).data
-
-        for wharton_booking in wharton_bookings:
-            booking_ids.add(wharton_booking["booking_id"])
-
+        url = f"{WHARTON_URL}{user.username}/reservations"
+        bookings = self.request("GET", url).json()["bookings"]
+
+        bookings = [booking for booking in bookings if datetime.datetime.strptime(booking["end"], "%Y-%m-%dT%H:%M:%S%z") >= timezone.localtime()]
```

Comment:
Might be cleaner if we can have a `serialize_reservation` function here to prettify the output (like what you did a few lines down) and then we can consolidate the two list comprehensions into one. Or similarly you could just not use the function and directly consolidate.